packages/core/http/core-http-common/src/versioning.ts

```diff
@@ -8,10 +8,16 @@
 
 /**
  * A Kibana HTTP API version
- * @note assumption that version will be monotonically increasing number where: version > 0.
+ *
+ * @note
+ * For public APIs: conforms to the Elastic API version specification APIs as a date string formatted as YYYY-MM-DD.
+ *
+ * @note
+ * For internal APIs: follow the convention of monotonic increasing integers.
+ *
  * @experimental
  */
-export type ApiVersion = `${number}`;
+export type ApiVersion = string;
 
 /** @internal */
 export const ELASTIC_HTTP_VERSION_HEADER = 'elastic-api-version' as const;
```

Comment:
nit: move to a const file. Even if we only have one exported item in the file, it makes it far easier to track changes